fastcrypto/src/groups/secp256k1.rs

```diff
@@ -0,0 +1,284 @@
+        Self(SecpProjectivePoint::default())
+    }
+
+    fn generator() -> Self {
+        Self(SecpProjectivePoint::GENERATOR)
+    }
+}
+
+impl Div<Scalar> for ProjectivePoint {
+    type Output = FastCryptoResult<Self>;
+
+    fn div(self, rhs: Scalar) -> Self::Output {
+        let inv = rhs.0.invert().unwrap();
+        Ok(Self(self.0.mul(inv)))
+    }
+}
+
+impl Mul<Scalar> for ProjectivePoint {
+    type Output = Self;
+
+    fn mul(self, rhs: Scalar) -> Self::Output {
+        Self(self.0.mul(rhs.0))
+    }
+}
+
+impl Sub for ProjectivePoint {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self(self.0.sub(rhs.0))
+    }
+}
+
+impl Add for ProjectivePoint {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(self.0.add(rhs.0))
+    }
+}
+
+impl Neg for ProjectivePoint {
+    type Output = Self;
+
+    fn neg(self) -> Self::Output {
+        Self(self.0.neg())
+    }
+}
+
+impl MultiScalarMul for ProjectivePoint {
+    fn multi_scalar_mul(scalars: &[Self::ScalarType], points: &[Self]) -> FastCryptoResult<Self> {
+        if scalars.len() != points.len() || scalars.is_empty() {
+            return Err(FastCryptoError::GeneralError(String::from(
+                "Invalid input: Scalars and points must have the same non-zero length",
+            )));
+        }
+
+        let mut result = SecpProjectivePoint::IDENTITY;
+
+        for (scalar, point) in scalars.iter().zip(points.iter()) {
+            if scalar.0.is_zero().into() || point.0 == SecpProjectivePoint::IDENTITY {
+                continue;
+            }
+
+            let scalar_mul = point.0.mul(scalar.0);
+
+            result += scalar_mul;
+        }
+
+        Ok(ProjectivePoint(result))
+    }
+}
+
+impl FromTrustedByteArray<PROJECTIVE_POINT_BYTE_LENGTH> for ProjectivePoint {
+    fn from_trusted_byte_array(
+        bytes: &[u8; PROJECTIVE_POINT_BYTE_LENGTH],
+    ) -> FastCryptoResult<Self> {
+        Ok(ProjectivePoint(
+            SecpProjectivePoint::from_bytes(bytes.as_slice().into()).unwrap(),
+        ))
+    }
+}
+
+impl ToFromByteArray<PROJECTIVE_POINT_BYTE_LENGTH> for ProjectivePoint {
+    fn from_byte_array(bytes: &[u8; PROJECTIVE_POINT_BYTE_LENGTH]) -> FastCryptoResult<Self> {
+        Self::from_trusted_byte_array(bytes)
+    }
+
+    fn to_byte_array(&self) -> [u8; PROJECTIVE_POINT_BYTE_LENGTH] {
+        self.0.to_bytes().into()
+    }
+}
+
+impl HashToGroupElement for ProjectivePoint {
+    fn hash_to_group_element(msg: &[u8]) -> Self {
+        let domain = "secp256k1_XMD:SHA-256_SSWU_RO_".as_bytes();
+        let mut u = [FieldElement::ZERO];
+        hash_to_field::<ExpandMsgXmd<Sha256>, FieldElement>(&[msg], &[domain], &mut u)
+            .expect("should never return error according to error cases described in ExpandMsgXmd");
+        Self(u[0].map_to_curve())
+    }
+}
+
+serialize_deserialize_with_to_from_byte_array!(ProjectivePoint);
+generate_bytes_representation!(
```

Comment:
I guess that you don't need this - we mainly use `XAsBytes` for external interfaces